quinn-interop/src/server.rs

```diff
@@ -99,28 +99,19 @@ async fn main() -> Result<()> {
                 continue;
             }
         };
-        let handshake_data = match new_conn.handshake_data().await {
-            Ok(x) => x,
-            Err(e) => {
-                error!("connection attempt failed: {}", e);
-                continue;
-            }
-        };
-        let alpn = String::from_utf8_lossy(
-            &handshake_data
-                .downcast_ref::<HandshakeData>()
-                .unwrap()
-                .protocol
-                .as_ref()
-                .unwrap()
-                .clone(),
-        )
-        .to_string();
-        trace!("New connection being attempted for {}", alpn);
-
         tokio::spawn(async move {
+            let handshake_data = match new_conn.handshake_data().await {
+                Ok(x) => x.downcast::<HandshakeData>().unwrap(),
+                Err(e) => {
+                    error!("connection attempt failed: {}", e);
+                    return;
+                }
+            };
+            let alpn = String::from_utf8_lossy(&handshake_data.protocol.as_ref().unwrap());
```

Comment:
Pre-existing issue, but it would be better to keep ALPN in the `Vec<u8>` domain instead of converting to `String`.